### DISCUSSION TITLE
Unreliable bltouch readings when creating a mesh

### DISCUSSION THREAD

**Discussion by joelferreira14** [mriscoc/Ender3V2S1#1021] [General]
Hey there!

I updated the firmware to the new version and I am getting some weird issues with the mesh.

Basically the bigger the mesh is, the bigger is the differences from the edges to the center. Pictures below.

Started with a 3x3 mesh
![PXL_20230813_090137607](https://github.com/mriscoc/Ender3V2S1/assets/48366483/46e9b821-c7e2-4bbc-b6da-1aee93b55dd5)

Then changed to a 5x5 and built another mesh
![PXL_20230813_090541823](https://github.com/mriscoc/Ender3V2S1/assets/48366483/c94f0d17-a6c4-4784-aab3-d47ae5a68ea2)

And lastly tryed the 9x9 Mesh
![PXL_20230813_091532173](https://github.com/mriscoc/Ender3V2S1/assets/48366483/50dd124c-eccc-444c-b759-091553df0434)

This must be some setting or option I am not seeing.

My printer is an ender 3 v2 with a sprite pro and bltouch. I was running the DEC 2022 firmware version just fine.
I have HS mode ON and Multiple probing set to 2.

**Comment by mriscoc** [comment]
Please use this special [Pronterface version](https://github.com/mriscoc/Pronterface-PFEd/releases) and send the code: 

```gcode
C851 F0
M500
```
and report back.

**Reply by jessedenhollander** [reply]
i have a similar issue with my ender 3 V2, but this did not fix it. are there any other F values i could use?

**Comment by joelferreira14** [comment]
Yup, that was it.

Fixed.
![PXL_20230813_185649266](https://github.com/mriscoc/Ender3V2S1/assets/48366483/67e1b410-55fd-499a-af56-9636545d88df)

Thank you Miguel

**Reply by mriscoc** [reply]
How is your multiple probe configured?

**Reply by joelferreira14** [reply]
The only thing I did was installing the firmware and then set the multiple probe on the printer (Advanced -> Probe settings -> Multiple Probing) to 2.

At the time when I was troubleshooting the issue, I tried to turn HS off and set the multiple probe to 0 and it did not change the mesh problem.

I have tried the precompiled firmware first (both may and the latest release). Once installed I would "Restore Defaults" on the printer just to be sure.

Currently running a custom compilation using configs from the "Professional Firmware Configurator".

I hope this helps.

**Reply by mriscoc** [reply]
The `C851 Fxxx` correction is specific for each printer/probe/probing settings, so from your pictures seems that `F0` is ok.

**Comment by dsmayo1** [comment]
I was experiencing the same issue as well (extreme negative values on the outer mesh). The C851 F0 command via pronterface seems to have fixed it. 

I have an Ender 3 V2 with a 422 board, upgraded with a bltouch and sprite pro extruder. 

The issue seems to have started with the May release and is present in the current Aug release.  The March 2023 release and earlier seem to not have the issue. 

Thank you for sharing a solution for this!

**Comment by bagsofcole** [comment]
I'm having the same issue with a near stock Ender 3 V2 Neo with CRTouch.  Can I use that commant wiht octoprint?

C851 F0
M500

**Reply by bagsofcole** [reply]
sorry, after reading a little more, looks like i will ne the special prontoface, but i have a Mac... I'll keep digging

**Reply by bagsofcole** [reply]
more sorries, it appears to have worked with octoprint too!

**Reply by SargonofAssyria** [reply]
I have an Ender 3V2 with a Sprite Pro / CR touch and I have the same problem with a UBL mesh that is totally off on both right and left side.
I tried the above C851 F0 and M500 that did not help. I also disabled the HS option, that did not help either.
I also tried a smaller mesh (4x4) but that also did not help.
Any other solutions?
